packages/voice/src/models/v1/helper.ts

```diff
@@ -0,0 +1,239 @@
+import {
+  ConnectConfProps,
+  ConnectMxpProps,
+  ConnectPstnProps,
+  ConnectSipProps,
+  ParkProps,
+  RunMenuProps,
+  StartRecordingOptions,
+  SvamlActionConnectConf,
+  SvamlActionConnectMxp,
+  SvamlActionConnectPstn,
+  SvamlActionConnectSip,
+  SvamlActionContinue,
+  SvamlActionHangup,
+  SvamlActionPark,
+  SvamlActionRunMenu,
+  SvamlInstructionAnswer,
+  SvamlInstructionPlayFiles,
+  SvamlInstructionSay,
+  SvamlInstructionSendDtmf,
+  SvamlInstructionSetCookie,
+  SvamlInstructionStartRecording,
+  SvamlInstructionStopRecording,
+} from './mod-svaml';
+import {
+  AceResponse,
+  AceSvamlAction, AceSvamlInstruction,
+  CallbackResponse,
+  IceResponse,
+  IceSvamlAction, IceSvamlInstruction,
+  PieResponse,
+  PieSvamlAction, PieSvamlInstruction,
+} from './mod-callbacks';
+import { TtsVoice } from './enums';
+
+class SvamletBuilder<A, I, R extends CallbackResponse<A, I>> {
+  protected action: A | undefined;
+  protected instructions: I[] | undefined;
+
+  setAction = (action: A) => {
+    this.action = action;
+    return this;
+  };
+  addInstruction = (instruction: I) => {
+    if (!this.instructions) {
+      this.instructions = [];
+    }
+    this.instructions.push(instruction);
+    return this;
+  };
+  build = (): R => {
+    return {
+      action: this.action,
+      instructions: this.instructions,
+    } as R;
+  };
+}
+
+export class AceSvamletBuilder extends SvamletBuilder<AceSvamlAction, AceSvamlInstruction, AceResponse> {}
+
+export class IceSvamletBuilder extends SvamletBuilder<IceSvamlAction, IceSvamlInstruction, IceResponse> {}
+
+export class PieSvamletBuilder extends SvamletBuilder<PieSvamlAction, PieSvamlInstruction, PieResponse> {}
+
+export const customCalloutHelper = {
+  formatIceResponse: (action?: IceSvamlAction, ...instructions: IceSvamlInstruction[]): string => {
+    const iceResponse: IceResponse = {
+      action,
+      instructions: instructions.length > 0 ? instructions : undefined,
```

Comment:
`SvamletBuilder` constructor do not initialize with a default value the `instructions` field
So, could we imagine having an exception thrown here ?

Reply:
No, instructions are optional. In this approach with rest parameters, if no instructions are provided, the serialization will produce an `instructions` property with an empty array, which is not what we want. 
In the `SvamletBuilder`, if there is an empty array, this is because the user specifically added this empty array, so I let it be serialized (honestly, I don't know the impact of having an empty array versus no property at all)